FreeRTOS-Plus/Demo/AWS/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/ShadowDemoMainExample.c

```diff
@@ -0,0 +1,758 @@
+            /* The received powerOn state is different from the one we retained before, so we switch them
+             * and set the flag. */
+            ulCurrentPowerOnState = ulNewState;
+
+            /* State change will be handled in main(), where we will publish a "reported"
+             * state to the device shadow. We do not do it here because we are inside of
+             * a callback from the MQTT library, so that we don't re-enter
+             * the MQTT library. */
+            stateChanged = true;
+        }
+    }
+    else
+    {
+        LogError( ( "No powerOn in json document!!" ) );
+        xUpdateDeltaReturn = pdFAIL;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvUpdateAcceptedHandler( MQTTPublishInfo_t * pxPublishInfo )
+{
+    char * pcOutValue = NULL;
+    uint32_t ulOutValueLength = 0U;
+    uint32_t ulReceivedToken = 0U;
+    JSONStatus_t result = JSONSuccess;
+
+    configASSERT( pxPublishInfo != NULL );
+    configASSERT( pxPublishInfo->pPayload != NULL );
+
+    LogInfo( ( "/update/accepted json payload:%s.", ( const char * ) pxPublishInfo->pPayload ) );
+
+    /* Handle the reported state with state change in /update/accepted topic.
+     * Thus we will retrieve the client token from the json document to see if
+     * it's the same one we sent with reported state on the /update topic.
+     * The payload will look similar to this:
+     *  {
+     *      "state": {
+     *          "reported": {
+     *          "powerOn": 1
+     *          }
+     *      },
+     *      "metadata": {
+     *          "reported": {
+     *          "powerOn": {
+     *              "timestamp": 1596573647
+     *          }
+     *          }
+     *      },
+     *      "version": 14698,
+     *      "timestamp": 1596573647,
+     *      "clientToken": "022485"
+     *  }
+     */
+
+    /* Make sure the payload is a valid json document. */
+    result = JSON_Validate( pxPublishInfo->pPayload,
+                            pxPublishInfo->payloadLength );
+
+    if( result == JSONSuccess )
+    {
+        /* Get clientToken from json documents. */
+        result = JSON_Search( ( char * ) pxPublishInfo->pPayload,
+                              pxPublishInfo->payloadLength,
+                              "clientToken",
+                              sizeof( "clientToken" ) - 1,
+                              '.',
+                              &pcOutValue,
+                              ( size_t * ) &ulOutValueLength );
+    }
+    else
+    {
+        LogError( ( "Invalid json documents !!" ) );
+    }
+
+    if( result == JSONSuccess )
+    {
+        LogInfo( ( "clientToken: %.*s", ulOutValueLength,
+                   pcOutValue ) );
+
+        /* Convert the code to an unsigned integer value. */
+        ulReceivedToken = ( uint32_t ) strtoul( pcOutValue, NULL, 10 );
+
+        LogInfo( ( "receivedToken:%d, clientToken:%u \r\n", ulReceivedToken, ulClientToken ) );
```

Comment:
Both ulReceivedToken and ulClientToken  are of type uint32_t and should use the same format specifier.

Reply:
sure